core/src/test/scala/org/apache/spark/serializer/UnsafeKryoSerializerSuite.scala

```diff
@@ -0,0 +1,28 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.serializer
+
+class UnsafeKryoSerializerSuite extends KryoSerializerSuite {
+
+  // This test suite should run all tests in KryoSerializerSuite with kryo unsafe.
+
+  override def beforeAll() {
+    super.beforeAll()
+    conf.set("spark.kryo.unsafe", "true")
```

Comment:
Have you checked that this runs in the right order w.r.t. the code that initializes a SparkContext in KryoSerializerSuite / SharedSparkContext? You may want to print sc.conf in a test. Otherwise, it would be better to put conf.set("spark.kryo.unsafe", "true") at the top-level inside this class. SharedSparkContext does create a SparkContext in its beforeAll, but you seem to editing the conf after that is done.